public/pages/workflow_detail/workflow_detail.test.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import {
+  RouteComponentProps,
+  Route,
+  Switch,
+  MemoryRouter,
+} from 'react-router-dom';
+import { WorkflowDetail } from './workflow_detail';
+import { WorkflowDetailRouterProps } from '../../pages';
+import '@testing-library/jest-dom';
+import { mockStore } from '../../../test/utils';
+
+jest.mock('../../services', () => {
+  const { mockCoreServices } = require('../../../test');
+  return {
+    ...jest.requireActual('../../services'),
+    ...mockCoreServices,
+  };
+});
+
+const renderWithRouter = (initialEntries: string[]) =>
+  render(
+    <Provider store={mockStore}>
+      <MemoryRouter initialEntries={initialEntries}>
+        <Switch>
+          <Route
+            path="/workflow/:workflowId"
+            render={(props: RouteComponentProps<WorkflowDetailRouterProps>) => (
+              <WorkflowDetail setActionMenu={jest.fn()} {...props} />
+            )}
+          />
+        </Switch>
+      </MemoryRouter>
+    </Provider>
+  );
+
+describe('WorkflowDetail', () => {
+  test('renders the page with workflowId parameter', () => {
+    const workflowId = '12345';
+    const { getAllByText, getByText, getByRole } = renderWithRouter([
+      `/workflow/${workflowId}`,
+    ]);
+
+    expect(getAllByText('test_workflow').length).toBeGreaterThan(0);
```

Comment:
Let's move things like IDs / names into constants instead of hardcoding. That makes it easier to reuse in other tests.